cmd/gb-vendor/purge.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"fmt"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/constabulary/gb"
+	"github.com/constabulary/gb/cmd"
+	"github.com/constabulary/gb/cmd/gb-vendor/vendor"
+)
+
+func init() {
+	registerCommand(PurgeCmd)
+}
+
+func parseImports(root string) (map[string]bool, error) {
+	pkgs := make(map[string]bool) // Set
+
+	var walkFn = func(path string, info os.FileInfo, err error) error {
+		if info.IsDir() {
+			return nil
+		}
+		if filepath.Ext(path) != ".go" { // Parse only go source files
+			return nil
+		}
+
+		fs := token.NewFileSet()
+		f, err := parser.ParseFile(fs, path, nil, parser.ImportsOnly)
+		if err != nil {
+			return err
+		}
+
+		for _, s := range f.Imports {
+			if !gb.Stdlib[s.Path.Value] {
+				pkgs[strings.Replace(s.Path.Value, "\"", "", -1)] = true
+			}
+		}
+		return nil
+	}
+	
+	err := filepath.Walk(root, walkFn)
+	return pkgs, err
+}
+
+var PurgeCmd = &cmd.Command{
+	Name:      "purge",
+	ShortDesc: "purges all unreferenced dependencies",
+	Run: func(ctx *gb.Context, args []string) error {
+		m, err := vendor.ReadManifest(manifestFile(ctx))
+		if err != nil {
+			return fmt.Errorf("could not load manifest: %v", err)
+		}
+
+		imports, err := parseImports(ctx.Projectdir())
+		if err != nil {
+			return fmt.Errorf("import could not be parsed: %v", err)
+		}
+
+		dependencies := make([]vendor.Dependency, len(m.Dependencies))
+		copy(dependencies, m.Dependencies)
+
+		for _, d := range dependencies {
+			if !imports[d.Importpath] {
+				dep, err := m.GetDependencyForImportpath(d.Importpath)
```

Comment:
Here is the case that I am concerned about

project/src/a/a.go imports
project/vendor/src/b/c/c.go
and 
projec/vendor/src/b/c/c.go imports
project/vendor/src/b

I can see two problems which this code as written. GetDependencyForImportPath won't match b/c. as b/ was the import path vendored. 

And if it did, then does the code remove b/c. or just b, or both ?